services/src/main/java/org/keycloak/services/resources/admin/AuthenticationManagementResource.java

```diff
@@ -459,7 +459,13 @@ public Response addExecutionToFlow(@PathParam("flowAlias") String flowAlias, Map
 
         AuthenticationExecutionModel execution = new AuthenticationExecutionModel();
         execution.setParentFlow(parentFlow.getId());
-        execution.setRequirement(AuthenticationExecutionModel.Requirement.DISABLED);
+
+        ConfigurableAuthenticatorFactory conf = (ConfigurableAuthenticatorFactory) f;
+        if (conf.getRequirementChoices().length == 1)
+            execution.setRequirement(conf.getRequirementChoices()[0]);
+        else
+            execution.setRequirement(AuthenticationExecutionModel.Requirement.DISABLED);
```

Comment:
@mabartos While at this, though the change would work for the case, when there would be exactly one Requirement available for the authentication model, it might still select nothing, if `Requirement.DISABLED` wouldn't be allowed requirement for that flow, wouldn't it? Or is the `Requirement.DISABLED` available for all the existing authentication flows?

Also, would be good to have a testcase checking this change IMHO.

Reply:
@iankko AFAIK each execution in authentication flow has either only `Requirement.REQUIRED`, or it always contains `Requirement.DISABLED`. Yes, I agree with the testcase. Thanks!